crates/turborepo-auth/src/lib.rs

```diff
@@ -44,15 +64,27 @@ where
     Ok(())
 }
 
+/// Login writes a token to disk at token_path. If a token is already present,
+/// we do not overwrite it and instead log that we found an existing token.
 pub async fn login<F>(
-    api_client: APIClient,
+    api_client: &impl Client,
     ui: &UI,
+    token_path: impl AsRef<Path>,
     mut set_token: F,
     login_url_configuration: &str,
 ) -> Result<()>
 where
     F: FnMut(&str) -> Result<()>,
 {
+    // Check if token exists first.
+    if let Ok(token) = std::fs::read_to_string(token_path) {
+        if let Ok(response) = api_client.get_user(&token).await {
+            println!("{}", ui.apply(BOLD.apply_to("Existing token found!")));
+            print_cli_authorized(&response.user.email, &ui);
+            return Ok(());
+        }
+    }
+
```

Comment:
Heya @Zertsov! You're reading a JSON file, and then passing the entire thing as the token. This will never work.

Reply:
🤦